mono/utils/mono-time.c

```diff
@@ -15,16 +15,25 @@
 #include <utils/mono-time.h>
 
 
-#define MTICKS_PER_SEC 10000000
+#define MTICKS_PER_SEC (10 * 1000 * 1000)
+
+gint64
+mono_msec_ticks (void)
+{
+	return mono_100ns_ticks () / 10 / 1000;
+}
 
 #ifdef HOST_WIN32
 #include <windows.h>
 
-guint32
-mono_msec_ticks (void)
+/* we get "error: implicit declaration of function 'GetTickCount64'" */
+ULONGLONG GetTickCount64(void);
```

Comment:
You should wrap this into a `#ifndef _MSC_VER` to make sure we only get this on cygwin/mingw, otherwise we'll get warnings on the VS build.